src/bundle.c

```diff
@@ -254,12 +255,14 @@ static gboolean casync_make_arch(const gchar *idxpath, const gchar *contentpath,
 		r_ptr_array_addv(iargs, casync_argvp, TRUE);
 	}
 	g_ptr_array_add(iargs, NULL);
+	istr = (gchar**) g_ptr_array_free(iargs, FALSE);
+	iargs = NULL; /* prevent autoptr clenaup */
 
 	/* Outer process calll */
 	g_ptr_array_add(args, g_strdup("fakeroot"));
 	g_ptr_array_add(args, g_strdup("sh"));
 	g_ptr_array_add(args, g_strdup("-c"));
-	g_ptr_array_add(args, g_strjoinv(" ", (gchar**) g_ptr_array_free(iargs, FALSE)));
+	g_ptr_array_add(args, g_strjoinv(" ", istr));
```

Comment:
Why not
```suggestion
	g_ptr_array_add(args, g_strjoinv(" ", iargs->pdata));
```
as we already have a NULL ptr at the end of `iargs`? This would avoid the complexity of the temporary `istr`.

Reply:
Indeed, this is way simpler. Fixed.